### ISSUE TITLE
Map remains in change camera mode (control key) after changing windows

### ISSUE THREAD

**Issue by bsudekum** [mapbox/mapbox-gl-js#3389]
**mapbox-gl-js version**: `v0.26.0`
- mac
- chrome 53.0.2785.143
### Steps to Trigger Behavior
1. Use the control key change the camera on the map
2. Quickly change windows with command + tab
3. Go back to chrome
4. Map remains in change camera mode even though the control key has been released
### Expected Behavior

The map would not be remain in change camera/pitch mode since the control key is not pressed.
### Actual Behavior

The map remains in change camera/pitch mode even though the control key is not pressed.

**Comment by andrewharvey** [comment]
I can reproduce this on Linux Chrome 55 without changing window.

1. Press and hold Ctrl
2. Press and hold Left Mouse button
3. Release Ctrl
(can still rotate/pitch which I would expect to be able to)
4. Release Left Mouse Button
(map still stuck in rotate/pitch even though both Ctrl and Left Mouse button are released.

**Comment by stevage** [comment]
Yeah, I get this on both Mac and Windows.

The good news is that you can "unstick" the key by doing the reverse:

1. Press and hold mouse button
2. Press and hold mouse Ctrl
3. Release mouse button
4. Release Ctrl

**Comment by andrewharvey** [comment]
Ugh @bsudekum, I submitted a PR to try to fix this, however I've just realised it doesn't address your exact report.

The way the DragRotateHandler works at the moment is if you let go of Ctrl/Command with the mouse still down, DragRotate is still active. I think this is how it should work. However at the moment if you have the mouse down and are drag rotating, and change tabs and then mouseup, when you change tabs back, even though the mouse is up the map is still rotating. That's the bug I've tried to fix.
